Issue #1125: Allow markdown formatting in "Description"

kuchtiak-ufal opened the pull request:
Now,

Based on the 

```
MarkdownConfig = {
    enabled: false,
    mathjax: false,
  };
```
definition user can use the  https://commonmark.org/ or for https://www.mathjax.org/ formatting in the description property.

Example:
Description Metadata text:

```
## Math Expression
___
* enabled: true
* mathjax: true
___
The [Demo](https://www.mathjax.org/#demo)

When $a \ne 0$, there are two solutions to $ax^2 + bx + c = 0$ and they are
$$x = {-b \pm \sqrt{b^2-4ac} \over 2a}.$$
```

Result:
<img width="1365" height="695" alt="Screenshot 2025-10-13 at 17 27 28" src="https://github.com/user-attachments/assets/5300bcae-fc83-455e-8c12-4acf88ddd26c" />

kuchtiak-ufal committed:
Issue 1125: render description field using MardownConfig

kuchtiak-ufal committed:
Merge branch 'clarin-v7' into issue_#1125

kuchtiak-ufal committed:
fixed unit test

kuchtiak-ufal committed:
resolve Copilot comments

kuchtiak-ufal commented on src/app/item-page/simple/field-components/clarin-description-item-field/clarin-description-item-field.component.ts:
```diff
@@ -24,11 +27,18 @@ export class ClarinDescriptionItemFieldComponent implements OnInit {
   validTextMetadata: string;
 
+  /**
+   * This variable will be true if {@link environment.markdown.enabled} is true.
```
Changed.

kuchtiak-ufal committed:
formatting

kuchtiak-ufal committed:
fix randomly failing test

kuchtiak-ufal committed:
UI components use the local.description.usemarkdown property to handle markdown text, in the Home page - Add <Markdown formatted text> toggle button to render markdown formatted text

amadulhaxxani committed:
Merge remote-tracking branch 'origin/clarin-v7' into issue_#1125

amadulhaxxani committed:
Add delay to cc-license spec assertion

Update the SubmissionSectionCcLicensesComponent spec to wait before asserting the getCcLicenseLink call. The test now uses a setTimeout(350) and the Jasmine done callback, calling fixture.detectChanges inside the timeout before verifying submissionCcLicenseUrlDataService.getCcLicenseLink was invoked with the expected Map. This change prevents timing-related flakiness by allowing async updates to complete before the assertion.

amadulhaxxani committed:
Add markdown preview toggle for description fields

Add markdown preview toggle for description fields

amadulhaxxani committed:
Detect nested markdown toggle and add test

Detect nested markdown toggle and add test

amadulhaxxani committed:
cache toggle visibility and normalize usemarkdown truthy value handling

fix(markdown-preview): cache toggle visibility and normalize usemarkdown truthy value handling

amadulhaxxani committed:
Conditionally render markdown description panel

Conditionally render markdown description panel

amadulhaxxani committed:
Disable MathJax for markdown

Disable MathJax for markdown

amadulhaxxani commented on config/config.yml:
```diff
@@ -171,3 +171,7 @@ languages:
+
+markdown:
+  enabled: true
+  mathjax: true
```
@kosarko 
seems like that’s a product decision

amadulhaxxani committed:
Clarify default allowlisted markdown fields

Clarify default allowlisted markdown fields

amadulhaxxani committed:
Extract markdown allow list and fix toggle

Move markdown description metadata allow list into a shared constant and replace inline copies across components. Improve markdown preview toggle behavior and accessibility: add aria-label/aria-pressed attributes in the template, add shouldBindMarkdownToggleVisibility() and clearMarkdownToggleVisibilityBinding() to avoid binding/ subscriptions when markdown is disabled, field is not a textarea, or field is readOnly, and ensure proper cleanup. Update unit tests to handle global appConfig markdown state and add cases for hiding the toggle when global markdown is disabled or when not editing.

amadulhaxxani committed:
Merge branch 'clarin-v7' into issue_#1125

amadulhaxxani committed:
Merge branch 'clarin-v7' into issue_#1125

amadulhaxxani committed:
chore: merge clarin-v7 into issue_#1125 and resolve import conflict
